src/main/java/com/epam/ta/reportportal/core/item/impl/UpdateTestItemHandlerImpl.java

```diff
@@ -353,4 +352,179 @@ private void verifyTestItem(TestItem item, Long id) throws BusinessRuleViolation
+				BAD_REQUEST_ERROR,
+				"Unable to change status on test item with children"
+		);
+		StatusEnum actualStatus = testItem.getItemResults().getStatus();
+		switch (actualStatus) {
+			case PASSED:
+				changeStatusFromPassed(testItem, providedStatus, user.getUserId(), projectDetails.getProjectId());
+				break;
+			case FAILED:
+				changeStatusFromFailed(testItem, providedStatus, user.getUserId(), projectDetails.getProjectId());
+				break;
+			case SKIPPED:
+				changeStatusFromSkipped(testItem, providedStatus, user.getUserId(), projectDetails.getProjectId());
+				break;
+			case INTERRUPTED:
+				changeStatusFromInterrupted(testItem, providedStatus, user.getUserId(), projectDetails.getProjectId());
+				break;
+			default:
+				throw new ReportPortalException(INCORRECT_REQUEST,
+						"actual status: " + actualStatus + " can not be changed to: " + providedStatus
+				);
+		}
+	}
+
+	private void changeStatusFromInterrupted(TestItem testItem, StatusEnum providedStatus, Long userId, Long projectId) {
+		expect(providedStatus, statusIn(SKIPPED, PASSED, FAILED)).verify(INCORRECT_REQUEST,
+				"actual status: " + testItem.getItemResults().getStatus() + " can not be changed to: " + providedStatus
+		);
+
+		TestItemActivityResource before = TO_ACTIVITY_RESOURCE.apply(testItem);
+		testItem.getItemResults().setStatus(providedStatus);
+
+		Optional<ItemAttribute> skippedIssueAttribute = itemAttributeRepository.findSystemAttributeByLaunchIdAndValue(testItem.getLaunch()
+				.getId(), SKIPPED_ISSUE_KEY);
+
+		if (FAILED.equals(providedStatus) || (SKIPPED.equals(providedStatus) && skippedIssueAttribute.isPresent()
+				&& skippedIssueAttribute.get().getValue().equals("true"))) {
+			setToInvestigateIssue(testItem, projectId);
+		}
+
+		testItemRepository.save(testItem);
+		messageBus.publishActivity(new TestItemStatusChangedEvent(before, TO_ACTIVITY_RESOURCE.apply(testItem), userId));
+
+		if (PASSED.equals(providedStatus)) {
+			changeStatusRecursively(testItem, userId);
+		}
+	}
+
+	private void changeStatusFromPassed(TestItem testItem, StatusEnum providedStatus, Long userId, Long projectId) {
+		expect(providedStatus, statusIn(SKIPPED, FAILED)).verify(INCORRECT_REQUEST,
+				"actual status: " + testItem.getItemResults().getStatus() + " can not be changed to: " + providedStatus
+		);
+
+		StatusEnum oldParentStatus = testItem.getParent().getItemResults().getStatus();
+		TestItemActivityResource before = TO_ACTIVITY_RESOURCE.apply(testItem);
+
+		Optional<ItemAttribute> skippedIssueAttribute = itemAttributeRepository.findSystemAttributeByLaunchIdAndValue(testItem.getLaunch()
+				.getId(), SKIPPED_ISSUE_KEY);
+
+		testItem.getItemResults().setStatus(providedStatus);
+		if (FAILED.equals(providedStatus) || (SKIPPED.equals(providedStatus) && skippedIssueAttribute.isPresent()
+				&& skippedIssueAttribute.get().getValue().equals("true"))) {
+			setToInvestigateIssue(testItem, projectId);
+		}
+
+		testItemRepository.save(testItem);
```

Comment:
we don't need to say this. as it is in transaction it will be saved after transaction commit